### PR TITLE
Merge `membrane-webrtc-js` into `ts-client-sdk`

### PR DESCRIPTION
**Pull request by mironiasty**
This PR is merging logic from https://github.com/jellyfish-dev/membrane-webrtc-js into this repository. So we can keep only one SDK for web (apart from React SDK - that can still be separate).

Next phase would be to generate new documentation for joined library and update/release new version of `ts-client-sdk`

> [!NOTE]  
> Commit history from previous repo is preserved using trick from https://gfscott.com/blog/merge-git-repos-and-keep-commit-history/

### CODE REVIEW

**Comment by mickel8** [comment]
@mironiasty Will `membrane-webrtc-js` be still published as a separate package on npmjs?

**Review by mickel8 — COMMENTED** [review]
@mironiasty Will membrane-webrtc-js be still published as a separate package on npmjs?

**Comment by mironiasty** [comment]
No. `membrane-webrtc-js` will be archived. All functionality from it will be available with `ts-client-sdk`.

File: e2e/app/src/vite-env.d.ts
Comment (karkakol) [inline e2e/app/src/vite-env.d.ts:1]: Why this file exists?

File: e2e/docker-compose-test.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ version: "3"
 
 services:
   jellyfish:
-    image: "ghcr.io/jellyfish-dev/jellyfish:${JELLYFISH_VERSION:-0.2.0}"
+    image: "ghcr.io/jellyfish-dev/jellyfish:${JELLYFISH_VERSION:-0.4.0}"
```
Comment (karkakol) [inline e2e/docker-compose-test.yaml:5]: I think it can be bumped to 0.4.2 or 0.5.0-rco

**Review by karkakol — APPROVED** [review]
Refer to my comments, then 👍

**Comment by karkakol** [comment]
One more thing, lets add some pre-commit hook to run prittier. It is annoying when some tests in github actions don't pass 🙏 . If not in this pr lets create task for it.

File: e2e/app/src/vite-env.d.ts
Comment (mironiasty) [reply e2e/app/src/vite-env.d.ts:1]: Idk. Maybe @kamil-stasiak can help
(source: https://github.com/jellyfish-dev/ts-client-sdk/commit/02726e909804c1f9f74d37cf96b84f419d86e387#diff-c2e96b5922a2839334d3d729952888e1b7e42d7158af4b03afdcc99b0332728d)

File: e2e/docker-compose-test.yaml (hunk 1 above)
Comment (mironiasty) [reply e2e/docker-compose-test.yaml:5]: I bumped it to 0.4.2. Let's use 0.5.0 when it is official

**Comment by mironiasty** [comment]
> One more thing, lets add some pre-commit hook to run prittier. It is annoying when some tests in github actions don't pass 🙏 . If not in this pr lets create task for it.

I will create task to unify checks across our repos, so each of them follow similar template

**Pull request merged**